src/main/java/ca/corefacility/bioinformatics/irida/database/changesets/AutomatedAnalysisToTemplate.java

```diff
@@ -0,0 +1,185 @@
+
+			//get the workflow identifiers
+			UUID assemblyId = assemblyWorkflow.getWorkflowIdentifier();
+			//get the default parameters
+			List<IridaWorkflowParameter> defaultAssemblyParams = assemblyWorkflow.getWorkflowDescription()
+					.getParameters();
+
+			//insert the assembly
+			insertWorkflow(jdbcTemplate, "Automated AssemblyAnnotation", true, assemblyId, "p.assemble_uploads=1",
+					defaultAssemblyParams);
+
+		} catch (IridaWorkflowNotFoundException e) {
+			logger.warn(
+					"Assembly workflow not found.  Automated assemblies will not be converted to analysis templates.");
+			//Note this will definitely happen in the galaxy CI tests as only SNVPhyl and a test workflow are configured.
+		}
+
+		//get the workflow information from the service
+		try {
+			sistrWorkflow = workflowsService.getDefaultWorkflowByType(BuiltInAnalysisTypes.SISTR_TYPING);
+
+			logger.debug("Updating automated SISTR project settings");
+			//get the workflow identifiers
+			UUID sistrId = sistrWorkflow.getWorkflowIdentifier();
+
+			//get the default params
+			List<IridaWorkflowParameter> defaultSistrParams = sistrWorkflow.getWorkflowDescription()
+					.getParameters();
+
+			//insert the sistr entries without metadata
+			insertWorkflow(jdbcTemplate, "Automated SISTR Typing", false, sistrId, "p.sistr_typing_uploads='AUTO'",
+					defaultSistrParams);
+			//insert the sistr entries with metadata
+			insertWorkflow(jdbcTemplate, "Automated SISTR Typing", true, sistrId,
+					"p.sistr_typing_uploads='AUTO_METADATA'", defaultSistrParams);
+		} catch (IridaWorkflowNotFoundException e) {
+			logger.warn("SISTR workflow not found.  Automated SISTR will not be converted to analysis templates.");
+			//Note this will definitely happen in the galaxy CI tests as only SNVPhyl and a test workflow are configured.
+		}
+
+		return new SqlStatement[0];
+	}
+
+	private void insertWorkflow(JdbcTemplate jdbcTemplate, String name, boolean updateSamples, UUID workflowId,
+			String where, List<IridaWorkflowParameter> params) {
+		/*
+		 * we're borrowing the 'automated' flag here to mark entries we need to add params to later.  at this point
+		 * there'll be nothing with a '1' in the automated flag.  we'll clear it later.
+		 */
+
+		//first get the project ids that have an automated submission
+		String idSql = "SELECT p.id FROM project p WHERE " + where;
+		List<Long> projectIds = jdbcTemplate.queryForList(idSql, Long.class);
+
+		//build the params for the insert submission
+		List<Object[]> queryParams = projectIds.stream()
+				.map(p -> {
+					return new Object[] { name, updateSamples, workflowId.toString(), p };
+				})
+				.collect(Collectors.toList());
+
+		//then insert the submisisons for each project
+		String submissionInsert = "INSERT INTO analysis_submission (DTYPE, name, created_date, priority, update_samples, workflow_id, submitter, submitted_project_id, automated) VALUES ('AnalysisSubmissionTemplate', ?, now(), 'LOW', ?, ?, 1, ?, 1)";
```

Comment:
In the past our stance has generally been to **not** add rows to the audit tables in cases that are complicated such as this one.  This is because it's quite hard to retroactively fully replicate what Envers does when it's creating revisions.  We would need to add `Revisions` entries, users, clients, timestamps, and most difficulty attaching the new `_AUD` entry to the *right* revision since it may have been created along with a project, etc.  With all that it would be easy to end up with the revisions and `_AUD`s in an inconsistent (or incorrect) state.

Given that stance, if you need to see when something like that was created, you can do it in the database backups.  The AUD records for that are still alive and well there.

Something we could do for posterity is add something to the `analysisDescription` field of the submission saying it was "translated from an old auto assembly on this date" etc. That way at least people can see where it came from.  How do you feel about that?

Reply:
I think that makes sense. I can see it being quite complicated to replicate everything that the audit tables would have entered. I think adding that message in the `analysisDescription` field is a good compromise.

Something else I was concerned about was whether or not new audit entries would still work if old ones weren't created? But, I think each entry is independent from the previous ones, so it won't make much of a difference.

Reply:
Added a description in 27ca2d17fd

Reply:
This works great. Thanks.